pkg/ottl/ottlfuncs/func_to_time_test.go

```diff
@@ -0,0 +1,276 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+package ottlfuncs
+
+import (
+	"context"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl"
+)
+
+func Test_toTsime(t *testing.T) {
+	tests := []struct {
+		name     string
+		time     ottl.StringGetter[interface{}]
+		format   ottl.StringGetter[interface{}]
+		expected interface{}
+	}{
+		{
+			name: "simple short form",
+			time: &ottl.StandardTypeGetter[interface{}, string]{
+				Getter: func(ctx context.Context, tCtx interface{}) (interface{}, error) {
+					return "2023-04-12", nil
+				},
+			},
+			format: &ottl.StandardTypeGetter[interface{}, string]{
+				Getter: func(ctx context.Context, tCtx interface{}) (interface{}, error) {
+					return "2006-01-02", nil
```

Comment:
I think it is because I've used other formats more, but after seeing these test cases I find Go's format logic very jarring.  I was definitely expecting the format to look like `"%Y-%m-%d"`.  Since many collector users are not Go users, I think if we can find a solid library that lets us use strptime format that will be preferred.